components/studio/bin/hab-studio.sh

```diff
@@ -679,11 +679,25 @@ enter_studio() {
     set -x
   fi
 
+  echo $studio_supervisor_start_command | $bb chroot "$HAB_STUDIO_ROOT" \
+    $studio_env_command -i $env $studio_run_command
+
+  trap stop_supervisor EXIT
+
   # Become the `chroot` process
-  exec $bb chroot "$HAB_STUDIO_ROOT" \
+  $bb chroot "$HAB_STUDIO_ROOT" \
     $studio_env_command -i $env $studio_enter_command $*
 }
 
+# **Internal** Run when an entered studio exits.
+stop_supervisor() {
+  lock_file="$HAB_STUDIO_ROOT/hab/sup/default/LOCK"
+
+  if [ -f $lock_file ]; then
+    $bb kill -9 $($bb cat $lock_file)
```

Comment:
Should we try to let the supervisor gracefully exit instead of forcing it with `-9` immediately?